meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/TestStressZkClient.java

```diff
@@ -1,24 +1,394 @@
+          Assert.fail("Should have failed due to bad expected version in set function.");
+        } catch (MetaClientException ex) {
+          Assert.assertEquals(ex.getClass().getName(),
+              "org.apache.helix.metaclient.exception.MetaClientBadVersionException");
+        }
+      }
+
+      zkMetaClient.delete(zkParentKey);
+      zkMetaClient.create(zkParentKey, "parent_node");
+
+      // Set with path to non-existent node - should fail
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        try {
+          zkMetaClient.set("/a/b/c", "test-should-fail", -1);
+          Assert.fail("Should have failed due to path to non-existent node");
+        } catch (Exception ignoredException) {
+        }
+      }
+
+      zkMetaClient.delete(zkParentKey);
+      zkMetaClient.create(zkParentKey, "parent_node");
+
+      // Set with invalid path - should fail
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        try {
+          zkMetaClient.set("a/b/c", "test-should-fail", -1);
+          Assert.fail("Should have failed due to invalid path - no leading /.");
+        } catch (Exception ignoredException) {
+        }
+      }
+    }
+
+    cleanUpTestNodes(zkParentKey);
+  }
+
+  @Test
+  public void testSetMultiNode() {
+    final String zkParentKey = "/stressZk_testSetMultiNode";
+    try (ZkMetaClient<String> zkMetaClient = createZkMetaClient()) {
+      zkMetaClient.connect();
+      zkMetaClient.create(zkParentKey, "parent_node");
+
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        zkMetaClient.create(zkParentKey + "/" + i, i);
+      }
+
+      // Set with no expected version
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        zkMetaClient.set(zkParentKey + "/" + i, "-" + i, -1);
+      }
+
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        String childKey = zkParentKey + "/" + i;
+        MetaClientInterface.Stat entryStat = zkMetaClient.exists(childKey);
+
+        Assert.assertEquals(zkMetaClient.get(childKey), "-" + i);
+        Assert.assertEquals(entryStat.getVersion(), 1);
+        Assert.assertEquals(entryStat.getEntryType().name(), PERSISTENT.name());
+      }
+
+      // Set with expected version
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        String childKey = zkParentKey + "/" + i;
+        zkMetaClient.set(childKey, String.valueOf(i), 1);
+        MetaClientInterface.Stat entryStat = zkMetaClient.exists(childKey);
+
+        Assert.assertEquals(zkMetaClient.get(childKey), String.valueOf(i));
+        Assert.assertEquals(entryStat.getVersion(), 2);
+      }
+    }
+    cleanUpTestNodes(zkParentKey);
+  }
+
+  @Test
+  public void testUpdateSingleNode() {
+    final String zkParentKey = "/stressZk_testUpdateSingleNode";
+    ZkMetaClientConfig config =
+        new ZkMetaClientConfig.ZkMetaClientConfigBuilder().setConnectionAddress(ZK_ADDR).build();
+    try (ZkMetaClient<Integer> zkMetaClient = new ZkMetaClient<>(config)) {
```

Comment:
I would recommend using a more complex node type here like DataRecord.

Reply:
Changed testUpdate for SingleNode and MultiNode to use datarecord